srv/salt/_modules/packagemanager.py

```diff
@@ -0,0 +1,274 @@
+from subprocess import Popen, PIPE
+from platform import linux_distribution
+import logging
+import os
+
+log = logging.getLogger(__name__)
+
+
+class PackageManager(object):
+
+    """
+    That is not the nativ salt module and is not ment to
+    replace it.
+    This module was created to react on packagemanagers
+    reboot advice.
+    """
```

Comment:
re 1) I guess I kept it there for other devs wondering why the hell we're reinvented the wheel.

re 2) yes, that is exactly the point where contributing to upstream will help us in the long run. lack of time prohibited me from doing that, yet. 

I'd *love* to replace this whole module with the pkg.* salt-builtins